skrifa/src/util.rs

```diff
@@ -0,0 +1,151 @@
+//! Internal helpers.
+
+use core::fmt;
+
+/// SmallVec like implementation but for fixed sized arrays whose
+/// size is only known at runtime.
+///
+/// Used to avoid allocations when possible for small or temporary
+/// arrays.
+///
+/// Consider taking on a smallvec dependency in the future.
+#[derive(Clone)]
```

Comment:
Might also be worth deriving `Default` and `PartialEq`/`Eq`?

Reply:
Good call on the comparison traits. Default is effectively meaningless however because it will always construct a zero length array and, unlike SmallVec, the length can’t be changed after construction.